vllm/engine/llm_engine.py

```diff
@@ -797,7 +829,23 @@ def step(self) -> List[RequestOutput]:
         """
         seq_group_metadata_list, scheduler_outputs = self.scheduler.schedule()
 
-        if not scheduler_outputs.is_empty():
+        if scheduler_outputs.is_empty():
+            output = []
+        elif self.parallel_config.sep_prompt_token:
+            all_outputs = self._run_stage_workers(
+                "execute_model",
+                prompt_stage=seq_group_metadata_list[0].is_prompt,
+                driver_kwargs={
+                    "seq_group_metadata_list": seq_group_metadata_list,
+                    "blocks_to_swap_in": scheduler_outputs.blocks_to_swap_in,
+                    "blocks_to_swap_out": scheduler_outputs.blocks_to_swap_out,
+                    "blocks_to_copy": scheduler_outputs.blocks_to_copy,
+                    "blocks_to_nw": scheduler_outputs.blocks_to_nw,
+                })
```

Comment:
@aashaka Thanks for the heads up! I will take a look at the changes soon!

Reply:
@GindaChen  

>  this means prompt workers cannot work concurrently with decode workers 

I do not quite understand this part. Since decode works are provisioned separately,  why can them run concurrently with prompt workers here? 

> because each time we only schedule one set of workers to run.

one set of workers means <prompt, worker>  pair?

What you are trying to say is currently it doesn't support <List<prompt>, List<workers>> to accelerate the prefill and decoding phase? (which requires model parallelism)